### PR TITLE
Add fstest.TestFS

### PR DESCRIPTION
**Pull request by cugu**
Nice package, thanks!

Go offers a function to test fs.FS implementations: https://tip.golang.org/pkg/testing/fstest/#TestFS. I added a test for this and fixed the revealed issues.

Sorry for not adding an emoji to the commit message 🙈

### CODE REVIEW

File: file.go [hunk 1]
```diff
@@ -37,29 +37,36 @@ func (f *file) Close() error {
...
+	// directory already exhausted
+	if n <= 0 && f.readDirPos >= len(f.entries) {
+		return nil, nil
 	}
 
-	if f.readDirPos == len(f.entries) {
-		return nil, io.EOF
+	// read till end
+	if n > 0 && f.readDirPos+n > len(f.entries) {
+		err = io.EOF
```
Comment (nlepage) [inline file.go:52]: I just double checked how the `os` implementation behaves and it seems that we should never return a non empty slice and EOF at the same time.

For example with 3 files and n=2, I get:
f.ReadDir(2) // returns []{file1, file2}, nil
f.ReadDir(2) // returns []{file3}, nil
f.ReadDir(2) // returns []{}, EOF

So I think you can group the first 2 tests like this:
```go
if f.readDirPos >= len(f.entries) {
	if n <= 0 {
		return nil, nil
	} else {
		return nil, io.EOF
	}
}
```

It should allow you to get rid of the named return parameters, which is better for readability.

File: file.go [hunk 2]
```diff
@@ -86,7 +122,7 @@ func (rf *rootFile) Size() int64 {
 }
 
 func (rf *rootFile) Mode() fs.FileMode {
-	return fs.FileMode(fs.ModeDir | 0755)
+	return fs.ModeDir | 0755
```
Comment (nlepage) [inline file.go:125]: :+1:

**Review by nlepage — CHANGES_REQUESTED** [review]
Hi @cugu 

Thanks a lot for your contribution.

I didn't know about `testing/fstest`! Too bad it isn't even mentioned in `io/fs` documentation...

I asked for a change in ReadDir's implementation, tell me if it's ok with you.

Otherwise LGTM :+1:

File: file.go (hunk 1 above)
Comment (cugu) [reply file.go]: Good point, I changed it.

**Review by nlepage — APPROVED** [review]
Thank you

**Comment by nlepage** [comment]
Release this in v1.0.4

**Comment by nlepage** [comment]
@all-contributors please add @cugu for tests and code

**Pull request merged**